npk/src/npk.rs

```diff
@@ -674,14 +709,14 @@ fn gen_pseudo_files(manifest: &Manifest) -> Result<NamedTempFile, Error> {
         let mut subdir = PathBuf::from("/");
         for dir in target.iter().skip(1) {
             subdir.push(dir);
-            add_directory(file, &subdir, mode)?;
+            add_directory(file, &subdir, mode).await?;
         }
     }
 
     Ok(pseudo_file_entries)
 }
 
-fn sign_hashes(key_pair: &Keypair, hashes_yaml: &str) -> String {
+async fn sign_hashes(key_pair: &Keypair, hashes_yaml: &str) -> String {
     let signature = key_pair.sign(hashes_yaml.as_bytes());
     let signature_base64 = base64::encode(signature);
```

Comment:
missing: `block_in_place`

Reply:
This happens in memory only (`hashes_yaml` is a `&str`). Do we want to block on that too or only on disk IO?